edelphi/src/main/java/fi/metatavu/edelphi/query/thesis/TimeSerieThesisQueryPageHandler.java

```diff
@@ -384,6 +388,13 @@ public void exportData(QueryExportContext exportContext) {
 
     }
   }
+
+  @Override
+  public ReportPageCommentProcessor exportComments(QueryPage queryPage, PanelStamp stamp, List<QueryReply> replies) {
+    Double maxX = getDoubleOptionValue(queryPage, getDefinedOption("time_serie.maxX"));
+    String axisXTitle = getStringOptionValue(queryPage, getDefinedOption("time_serie.xAxisTitle"));
+    return new TimeSerieReportPageCommentProcessor(stamp, queryPage, new HashMap<>(), maxX, axisXTitle);
+  }
   
   @Override
   protected void renderReport(PageRequestContext requestContext, QueryPage queryPage) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 23 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)